addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -6958,10 +7055,10 @@ registry.ImageTools = ImageHandlerOption.extend({
             newColors = oldColors.map((color, i) => color !== null ? this._getCSSColorValue(`o-color-${(i + 1)}`) : null);
         }
         newColors.forEach((color, i) => shape = shape.replace(new RegExp(oldColors[i], 'g'), this._getCSSColorValue(color)));
-        await this._writeShape(shape);
         if (save) {
-            img.dataset.shapeColors = newColors.join(';');
+            this.imageOptions.shapeColors = newColors.join(';');
         }
+        await this._writeShape(shape);
```

Comment:
This line has been moved to take a potential change of `this.imageOptions.shapeColors` into account at the registry update.